presto-main/src/test/java/com/facebook/presto/server/remotetask/TestHttpRemoteTask.java

```diff
@@ -0,0 +1,237 @@
+import io.airlift.units.Duration;
+import org.testng.annotations.Test;
+
+import java.net.URI;
+import java.nio.charset.StandardCharsets;
+import java.util.concurrent.CompletableFuture;
+import java.util.concurrent.ExecutionException;
+import java.util.concurrent.atomic.AtomicLong;
+import java.util.function.BiConsumer;
+
+import static com.facebook.presto.SessionTestUtils.TEST_SESSION;
+import static com.facebook.presto.client.PrestoHeaders.PRESTO_MAX_WAIT;
+import static com.facebook.presto.client.PrestoHeaders.PRESTO_TASK_INSTANCE_ID;
+import static io.airlift.units.Duration.nanosSince;
+import static java.lang.String.format;
+import static java.util.concurrent.TimeUnit.MILLISECONDS;
+import static java.util.concurrent.TimeUnit.SECONDS;
+import static javax.ws.rs.core.HttpHeaders.CONTENT_TYPE;
+import static org.testng.Assert.assertTrue;
+
+public class TestHttpRemoteTask
+{
+    private final CompletableFuture<Void> testComplete = new CompletableFuture<>();
+
+    @Test
+    public void testRemoteTaskMismatch()
+            throws InterruptedException, ExecutionException
+    {
+        Duration idleTimeout = new Duration(3, SECONDS);
+        Duration failTimeout = new Duration(20, SECONDS);
+
+        JsonCodec<TaskStatus> taskStatusCodec = JsonCodec.jsonCodec(TaskStatus.class);
+        JsonCodec<TaskInfo> taskInfoCodec = JsonCodec.jsonCodec(TaskInfo.class);
+        TaskManagerConfig taskManagerConfig = new TaskManagerConfig();
+
+        // Shorten status refresh wait and info update interval so that we can have a shorter test timeout
+        taskManagerConfig.setStatusRefreshMaxWait(new Duration(idleTimeout.roundTo(MILLISECONDS) / 100, MILLISECONDS));
+        taskManagerConfig.setInfoUpdateInterval(new Duration(idleTimeout.roundTo(MILLISECONDS) / 10, MILLISECONDS));
+
+        AtomicLong lastActivity = new AtomicLong();
+        HttpProcessor httpProcessor = new HttpProcessor(taskStatusCodec, taskInfoCodec, lastActivity);
+        TestingHttpClient testingHttpClient = new TestingHttpClient(httpProcessor);
+
+        HttpRemoteTaskFactory httpRemoteTaskFactory = new HttpRemoteTaskFactory(
+                new QueryManagerConfig(),
+                taskManagerConfig,
+                testingHttpClient,
+                new TestSqlTaskManager.MockLocationFactory(),
+                taskStatusCodec,
+                taskInfoCodec,
+                JsonCodec.jsonCodec(TaskUpdateRequest.class),
+                new RemoteTaskStats());
+        RemoteTask remoteTask = httpRemoteTaskFactory.createRemoteTask(
+                TEST_SESSION,
+                new TaskId("test", 1, 2),
+                new PrestoNode("node-id", URI.create("http://192.0.1.2"), new NodeVersion("version"), false),
+                TaskTestUtils.PLAN_FRAGMENT,
+                ImmutableMultimap.of(),
+                OutputBuffers.createInitialEmptyOutputBuffers(OutputBuffers.BufferType.BROADCAST),
+                new NodeTaskMap.PartitionedSplitCountTracker(i -> {
```

Comment:
put the curlies on the same line like this `new PartitionedSplitCountTracker(delta -> { })`